aries_cloudagent/connections/models/diddoc/util.py

```diff
@@ -49,20 +49,34 @@ def canon_did(uri: str) -> str:
         ValueError: for invalid input.
 
     """
+    error_msg = 'Bad specification {} does not correspond to a DID'.format(uri)
+
+    uri_aux = uri.split(':')
+    if len(uri_aux) > 1:
+        uri = uri_aux[-1]
+        if not '{}:{}'.format(uri_aux[0], uri_aux[1]) in AVAILABLE_DIDS:
+            error_msg = 'Bad specification with DID {}. ' \
+                        'Method {} not available'.format(uri, uri_aux[1])
+            uri = ''
```

Comment:
We have a more complete DID and DID URL parsing solution in `aries_cloudagent.resolver.did `.